external/upstream/gauxc/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+if(${ENABLE_gauxc})
+    if(NOT "${gauxc_DIR}" STREQUAL "")
+        if("${IntegratorXX_DIR}" STREQUAL "")
+            set(IntegratorXX_DIR ${gauxc_DIR}/../IntegratorXX)
+        endif()
+        set(ExchCXX_DIR ${gauxc_DIR}/../ExchCXX)
+    endif()
+
+    get_property(BLAS_LIBRARIES TARGET lapk PROPERTY INTERFACE_LINK_LIBRARIES)
+
+    find_package(gauxc CONFIG) 
+
+    if(${gauxc_FOUND})
+        #== check if GPU-enabled GauXC is found, if GPU is specified ==#
+        get_property(_gpu TARGET gauxc::gauxc PROPERTY GAUXC_HAS_DEVICE)
+        
+        if(${gauxc_ENABLE_GPU} AND (NOT ${_gpu}))
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned on, but selected GauXC install is not built for GPU support!")
+        elseif((NOT ${gauxc_ENABLE_GPU}) AND ${_gpu})
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned off, but selected GauXC install is built for GPU support!")
```

Comment:
maybe I'm forgetting a conversation, but gauxc_ENABLE_GPU=OFF running on GauXC that supports CPU _or_ GPU sounds fine -- just tell it to run in CPU mode.

Reply:
We did indeed have a previous conversation that touched on this. The suggestion by @wavefunction91 in that discussion was that "if you set `gauxc_ENABLE_GPU`, you should check `GAUXC_HAS_DEVICE` is consistent in the discovered target", which is what we do here (in that we check that they are the same). Perhaps what you suggest here could also be interpreted as consistent with what @wavefunction91 suggested, as well, in which case I'm happy to change this up.

Reply:
Unless there's something peculiar about gauze where a gpu version doesn't run well on cpu I think change it. As it stands, someone with the default gauxcenablegpu setting of off who detected a cpu/gpu gauze install would be caught be this check. Lmk if I've interpreted wrong.

Reply:
You're interpretation is indeed correct, and that's a very valid point. I'll change this!

Reply:
Done! I switched the error here (and the analogous error for Magma support) to warnings, just in case someone wanted GPU support for the Psi4/GauXC interface and build GauXC as such, but forgot to switch it on Psi4-side.